diff --git a/helpers.cpp b/helpers.cpp
index df51221..24cda89 100644
--- a/helpers.cpp
+++ b/helpers.cpp
@@ -271,27 +271,13 @@ string cHelpers::GetEventsIntern(string wantedChannels, string when) {
 string cHelpers::DelRecording(cRecording * recording) {
 
 	cString FileName = recording->FileName();
-
-	if (cCutter::Active(recording->FileName())) {
-		cCutter::Stop();
-		recording = Recordings.GetByName(FileName); // cCutter::Stop() might have deleted it if it was the edited version
-		// we continue with the code below even if recording is NULL,
-		// in order to have the menu updated etc.
-	}
+	RecordingsHandler.Del(FileName);
 
 	if (cReplayControl::NowReplaying()
 			&& strcmp(cReplayControl::NowReplaying(), FileName) == 0) {
 		cControl::Shutdown();
 	}
 
-	if (!recording || recording->Delete()) {
-		cReplayControl::ClearLastReplayed(FileName);
-		Recordings.DelByName(FileName);
-#if VDRVERSNUM > 10727
-		cVideoDiskUsage::ForceCheck();
-#endif
-	}
-
 	return "START\r\nEND\r\n";
 }
 
