--- a/iptv/vlc2iptv	2014-03-09 06:20:00.000000000 +0400
+++ b/iptv/vlc2iptv	2014-03-12 00:05:28.106155252 +0400
@@ -52,11 +52,11 @@
     [ ! -e "$CHANNELS_CONF" ] && \
       exit_with_error "channels.conf not found ($CHANNELS_CONF)"
 
-    local CHANNEL_RECORD=`grep "[:]S=[10][|]P=[10][|]F=EXT[|]U=vlc2iptv[|]A=$PARAMETER[:]I" $CHANNELS_CONF`
+    local CHANNEL_RECORD="`grep "[:]S=[10][|]P=[10][|]F=EXT[|]U=vlc2iptv[|]A=$PARAMETER[:]I" $CHANNELS_CONF`"
     [ -z "$CHANNEL_RECORD" ] && \
       exit_with_error "no iptv channel with parameter $PARAMETER found"
 
-    CHANNEL_NAME=`echo $CHANNEL_RECORD | awk "-F[;,:]" '{print $1}'`
+    CHANNEL_NAME="`echo $CHANNEL_RECORD | awk "-F[;,:]" '{print $1}'`"
     VPID=`echo $CHANNEL_RECORD | awk -F: '{print $6}'`
     APID=`echo $CHANNEL_RECORD | awk -F: '{print $7}'`
     SPID=0
@@ -68,7 +68,7 @@
 
     [ ! -e $CHANNEL_SETTINGS_FILE ] && \
       exit_with_error "No vlc input configuration for channel '$CHANNEL_NAME'"
-    . $CHANNEL_SETTINGS_FILE
+    . "$CHANNEL_SETTINGS_FILE"
     
     [ -z "$URL" ] && \
       exit_with_error "No URL specified for channel '$CHANNEL_NAME'"
