diff --git a/device-tmpl.cpp b/device-tmpl.cpp
index 158324b..035d368 100644
--- a/device-tmpl.cpp
+++ b/device-tmpl.cpp
@@ -36,7 +36,6 @@ private:
   bool ScActive(void);
 
 protected:
-  virtual bool Ready(void);
   virtual bool SetPid(cPidHandle *Handle, int Type, bool On);
   virtual bool SetChannelDevice(const cChannel *Channel, bool LiveView);
   virtual bool OpenDvr(void);
@@ -141,12 +140,6 @@ bool SCDEVICE::HasCi(void)
   return sCCIAdapter || hwciadapter;
 }
 
-bool SCDEVICE::Ready(void)
-{
-  return (sCCIAdapter ? sCCIAdapter->Ready() : true) &&
-         (hwciadapter ? hwciadapter->Ready() : true);
-}
-
 bool SCDEVICE::SetPid(cPidHandle *Handle, int Type, bool On)
 {
   DEBUGLOG("%s: on=%d", __FUNCTION__, On);
